plugin/route53/setup.go

```diff
@@ -57,32 +60,35 @@ func setup(c *caddy.Controller, f func(*credentials.Credentials) route53iface.Ro
 					return c.Errf("invalid access key '%v'", v)
 				}
 				credential = credentials.NewStaticCredentials(v[0], v[1], "")
+			case "upstream":
```

Comment:
can you copy how the etcd plugin does this? i.e. just
~~~
case "upstream":
					args := c.RemainingArgs()
					if len(args) == 0 {
						return nil, false, c.ArgErr()
					}
					u, err := upstream.New(args)
					if err != nil {
						return nil, false, err
					}
					etc.Upstream = u
~~~
(but then of course adjusted to this plugin)

Reply:
hmm, it is optional, that's a bug in etcd's setup, #2099 filed.